File: search-a-2d-matrix/search-a-2d-matrix.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        '''
        Given a 2D list and a targeted value, returns if the target is in the array
        Returns true if in the matrix, returns false otherwise
        '''
        for row in range(0, len(matrix), 1):
            if row == (len(matrix) - 1) or (matrix[row][0] <= target and matrix[row+1][0] > target):
                for col in range(0, len(matrix[row]), 1):
                    if matrix[row][col] == target:
                        return True
                break
        return False

File: search-a-2d-matrix/search-a-2d-matrix.java
class Solution 
{
    public boolean searchMatrix(int[][] matrix, int target) 
    {
        int width = matrix[0].length;
        int height = matrix.length;

        int low = 0, hi = (width * height) - 1;
        while (low <= hi)
        {
            int mid = (low + hi) / 2;
            int row = mid / width;
            int col = mid % width;
            
            if (matrix[row][col] == target)
                return true;
            else if (matrix[row][col] < target)
                low = mid+1;
            else
                hi = mid-1;
        }
        return false;
    }
}